<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />




<title>Figure 7</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-1.1/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-1.1/highlight.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 60px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 65px;
  margin-top: -65px;
}

.section h2 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h3 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h4 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h5 {
  padding-top: 65px;
  margin-top: -65px;
}
.section h6 {
  padding-top: 65px;
  margin-top: -65px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>






<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Rictor/TORC2 mediates gut-to-brain signaling in the regulation of phenotypic plasticity in C.elegans</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    home
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="about.html">raw data</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Figures
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Figure_1.html">Figure 1</a>
    </li>
    <li>
      <a href="Figure_2.html">Figure 2</a>
    </li>
    <li>
      <a href="Figure_3.html">Figure 3</a>
    </li>
    <li>
      <a href="Figure_4.html">Figure 4</a>
    </li>
    <li>
      <a href="Figure_5.html">Figure 5</a>
    </li>
    <li>
      <a href="Figure_6.html">Figure 6</a>
    </li>
    <li>
      <a href="Figure_7.html">Figure 7</a>
    </li>
    <li>
      <a href="Figure_8.html">Figure 8</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Figure 7</h1>

</div>


<div id="identification-of-two-qtl-for-27-dauer-formation" class="section level1 tabset">
<h1><strong>Identification of two QTL for 27º dauer formation</strong></h1>
<div id="a" class="section level2">
<h2>7A</h2>
<pre class="r"><code>filepath&lt;-file.path(pathname,&#39;extdata/7A_CSSdata_allbinary.csv&#39;)
strains&lt;- c(&quot;N2&quot;,&quot;CB4856&quot;,&quot;CSSI&quot;,&quot;CSSII&quot;,&quot;CSSIII&quot;,&quot;CSSIV&quot;,&quot;CSSV&quot;,&quot;CSSX&quot;)
foods &lt;- &quot;OP50&quot;
CSS&lt;-read.csv(filepath, header=TRUE) %&gt;% format_dauer(p.dauer = &quot;exclude&quot;) %&gt;% dplyr::filter(n &gt; 9)</code></pre>
<pre class="r"><code>lm &lt;- CSS %&gt;% dauer_ANOVA()
glmm.locus &lt;- glmer(CSS,formula = cbind(dauer,non) ~ gtChrI + gtChrII + gtChrIII + gtChrIV + gtChrV + gtChrX + (1|plateID) + (1|strainDate), family = binomial(link = logit),control=glmerControl(optimizer=&quot;bobyqa&quot;))
contrasts&lt;-dauergut::dunnett_contrasts(lm, ref.index = 1, &quot;genotype&quot;)
plot.contrasts&lt;-c(&quot;&quot;,contrasts$prange[1:7])

stan.glmm &lt;- CSS %&gt;% run_dauer_stan
mixed&lt;- stan.glmm %&gt;% getStan_CIs(type = &quot;dauer&quot;) #CIs

(p&lt;-dauergut::plot_CIs(CSS, title=&#39;CB4856 Chromosome II contains 1 or more dauer QTLs&#39;, plot.contrasts, ypos = 1.075, type = &quot;dauer&quot;, labels = strains))</code></pre>
<p><img src="Figure_7_files/figure-html/CSS%20glmm-1.png" width="672" /></p>
<p><br> </br></p>
<p>7A. Chromosome substitution strains identify a chromosome II QTL for 27º dauer formation. Dauers formed by chromosome-substituted strains at 27°C. Each dot indicates the proportion of dauers formed in a single assay. Horizontal bar indicates median. Light gray thin and thick vertical bars at right indicate Bayesian 95% and 75% credible intervals, respectively. Numbers in parentheses below indicate the number of independent assays with at least 11 animals each. Shown P-values are with respect to the Bristol N2 strain unless indicated otherwise; <code>***</code> - different from Bristol N2 at P&lt;0.001 (ANOVA with Dunnett-type multivariate-t post-hoc adjustment).</p>
<pre class="r"><code>library(sjPlot)
sjt.lm(lm, depvar.labels = &quot;proportion of dauers&quot;, show.se = TRUE)</code></pre>
<table style="border-collapse:collapse; border:none;border-bottom:double;">
<td style="padding:0.2cm; border-top:double;">
 
</td>
<td style="border-bottom:1px solid; padding-left:0.5em; padding-right:0.5em; border-top:double;">
 
</td>
<td style="padding:0.2cm; text-align:center; border-bottom:1px solid; border-top:double;" colspan="4">
proportion of dauers
</td>
</tr>
<tr>
<td style="padding:0.2cm; font-style:italic;">
 
</td>
<td style="padding-left:0.5em; padding-right:0.5em; font-style:italic;">
 
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
B
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
CI
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
std. Error
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
p
</td>
</tr>
<tr>
<td style="padding:0.2cm; border-top:1px solid; text-align:left;">
(Intercept)
</td>
<td style="padding-left:0.5em; padding-right:0.5em; border-top:1px solid; ">
 
</td>
<td style="padding:0.2cm; text-align:center; border-top:1px solid; ">
0.38
</td>
<td style="padding:0.2cm; text-align:center; border-top:1px solid; ">
0.24 – 0.51
</td>
<td style="padding:0.2cm; text-align:center; border-top:1px solid; ">
0.07
</td>
<td style="padding:0.2cm; text-align:center; border-top:1px solid; ">
&lt;.001
</td>
</tr>
<tr>
<td style="padding:0.2cm;" colspan="6">
genotype
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CB4856
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.36
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.54 – -0.18
</td>
<td style="padding:0.2cm; text-align:center; ">
0.09
</td>
<td style="padding:0.2cm; text-align:center; ">
&lt;.001
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CSSI
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
0.06
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.15 – 0.28
</td>
<td style="padding:0.2cm; text-align:center; ">
0.11
</td>
<td style="padding:0.2cm; text-align:center; ">
.552
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CSSII
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.29
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.48 – -0.11
</td>
<td style="padding:0.2cm; text-align:center; ">
0.09
</td>
<td style="padding:0.2cm; text-align:center; ">
.003
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CSSIII
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.06
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.26 – 0.13
</td>
<td style="padding:0.2cm; text-align:center; ">
0.10
</td>
<td style="padding:0.2cm; text-align:center; ">
.517
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CSSIV
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.14
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.33 – 0.06
</td>
<td style="padding:0.2cm; text-align:center; ">
0.10
</td>
<td style="padding:0.2cm; text-align:center; ">
.158
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CSSV
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
0.13
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.08 – 0.35
</td>
<td style="padding:0.2cm; text-align:center; ">
0.11
</td>
<td style="padding:0.2cm; text-align:center; ">
.218
</td>
</tr>
<tr>
<td style="font-style:italic; padding:0.2cm; padding-left:0.6cm; text-align:left;">
CSSX
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.11
</td>
<td style="padding:0.2cm; text-align:center; ">
-0.30 – 0.09
</td>
<td style="padding:0.2cm; text-align:center; ">
0.10
</td>
<td style="padding:0.2cm; text-align:center; ">
.272
</td>
</tr>
<tr>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left; border-top:1px solid;">
Observations
</td>
<td style="padding-left:0.5em; padding-right:0.5em; border-top:1px solid;">
 
</td>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:center; border-top:1px solid;" colspan="4">
47
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
R<sup>2</sup> / adj. R<sup>2</sup>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
.522 / .436
</td>
</tr>
</table>
<pre class="r"><code>sjt.glmer(glmm.locus, depvar.labels = &quot;proportion of dauers&quot;, show.se = TRUE)</code></pre>
<table style="border-collapse:collapse; border:none;border-bottom:double;">
<tr>
<td style="padding:0.2cm; border-top:double;">
 
</td>
<td style="border-bottom:1px solid; padding-left:0.5em; padding-right:0.5em; border-top:double;">
 
</td>
<td style="padding:0.2cm; text-align:center; border-bottom:1px solid; border-top:double;" colspan="4">
proportion of dauers
</td>
</tr>
<tr>
<td style="padding:0.2cm; font-style:italic;">
 
</td>
<td style="padding-left:0.5em; padding-right:0.5em; font-style:italic;">
 
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
Odds Ratio
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
CI
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
std. Error
</td>
<td style="padding:0.2cm; text-align:center; font-style:italic; ">
p
</td>
</tr>
<tr>
<td colspan="6" style="padding:0.2cm; text-align:left; border-top:1px solid; font-weight:bold; text-align:left;">
Fixed Parts
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
(Intercept)
</td>
<td style="padding-left:0.5em; padding-right:0.5em; ">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
0.01
</td>
<td style="padding:0.2cm; text-align:center; ">
0.00 – 0.05
</td>
<td style="padding:0.2cm; text-align:center; ">
0.01
</td>
<td style="padding:0.2cm; text-align:center; ">
&lt;.001
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
gtChrI (N2)
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
0.87
</td>
<td style="padding:0.2cm; text-align:center; ">
0.30 – 2.49
</td>
<td style="padding:0.2cm; text-align:center; ">
0.47
</td>
<td style="padding:0.2cm; text-align:center; ">
.798
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
gtChrII (N2)
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
13.34
</td>
<td style="padding:0.2cm; text-align:center; ">
4.41 – 40.33
</td>
<td style="padding:0.2cm; text-align:center; ">
7.53
</td>
<td style="padding:0.2cm; text-align:center; ">
&lt;.001
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
gtChrIII (N2)
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
1.62
</td>
<td style="padding:0.2cm; text-align:center; ">
0.59 – 4.40
</td>
<td style="padding:0.2cm; text-align:center; ">
0.83
</td>
<td style="padding:0.2cm; text-align:center; ">
.347
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
gtChrIV (N2)
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
2.52
</td>
<td style="padding:0.2cm; text-align:center; ">
0.92 – 6.90
</td>
<td style="padding:0.2cm; text-align:center; ">
1.30
</td>
<td style="padding:0.2cm; text-align:center; ">
.072
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
gtChrV (N2)
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
0.64
</td>
<td style="padding:0.2cm; text-align:center; ">
0.20 – 2.06
</td>
<td style="padding:0.2cm; text-align:center; ">
0.38
</td>
<td style="padding:0.2cm; text-align:center; ">
.451
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left;">
gtChrX (N2)
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; ">
2.83
</td>
<td style="padding:0.2cm; text-align:center; ">
0.98 – 8.16
</td>
<td style="padding:0.2cm; text-align:center; ">
1.53
</td>
<td style="padding:0.2cm; text-align:center; ">
.055
</td>
</tr>
<tr>
<td colspan="6" style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left; font-weight:bold; text-align:left; padding-top:0.5em;">
Random Parts
</td>
</tr>
<tr>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;">
τ<sub>00, plateID</sub>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
0.286
</td>
</tr>
<tr>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;">
τ<sub>00, strainDate</sub>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
0.581
</td>
</tr>
<tr>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;">
N<sub>plateID</sub>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
45
</td>
</tr>
<tr>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left;">
N<sub>strainDate</sub>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
23
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
ICC<sub>plateID</sub>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
0.069
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
ICC<sub>strainDate</sub>
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
0.140
</td>
</tr>
<tr>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:left; border-top:1px solid;">
Observations
</td>
<td style="padding-left:0.5em; padding-right:0.5em; border-top:1px solid;">
 
</td>
<td style="padding:0.2cm; padding-top:0.1cm; padding-bottom:0.1cm; text-align:center; border-top:1px solid;" colspan="4">
47
</td>
</tr>
<tr>
<td style="padding:0.2cm; text-align:left; padding-top:0.1cm; padding-bottom:0.1cm;">
Deviance
</td>
<td style="padding-left:0.5em; padding-right:0.5em;">
 
</td>
<td style="padding:0.2cm; text-align:center; padding-top:0.1cm; padding-bottom:0.1cm;" colspan="4">
23.557
</td>
</tr>
</table>
<pre class="r"><code>knitr::kable(drop1(glmm.locus, test = &quot;Chisq&quot;))</code></pre>
<table>
<thead>
<tr class="header">
<th></th>
<th align="right">Df</th>
<th align="right">AIC</th>
<th align="right">LRT</th>
<th align="right">Pr(Chi)</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><none></td>
<td align="right">NA</td>
<td align="right">255.8060</td>
<td align="right">NA</td>
<td align="right">NA</td>
</tr>
<tr class="even">
<td>gtChrI</td>
<td align="right">1</td>
<td align="right">253.8717</td>
<td align="right">0.0656622</td>
<td align="right">0.7977605</td>
</tr>
<tr class="odd">
<td>gtChrII</td>
<td align="right">1</td>
<td align="right">269.9765</td>
<td align="right">16.1704903</td>
<td align="right">0.0000579</td>
</tr>
<tr class="even">
<td>gtChrIII</td>
<td align="right">1</td>
<td align="right">254.6654</td>
<td align="right">0.8593609</td>
<td align="right">0.3539176</td>
</tr>
<tr class="odd">
<td>gtChrIV</td>
<td align="right">1</td>
<td align="right">256.7430</td>
<td align="right">2.9369918</td>
<td align="right">0.0865718</td>
</tr>
<tr class="even">
<td>gtChrV</td>
<td align="right">1</td>
<td align="right">254.3687</td>
<td align="right">0.5626294</td>
<td align="right">0.4532027</td>
</tr>
<tr class="odd">
<td>gtChrX</td>
<td align="right">1</td>
<td align="right">257.5262</td>
<td align="right">3.7201793</td>
<td align="right">0.0537586</td>
</tr>
</tbody>
</table>
</div>
<div id="s4a" class="section level2">
<h2>S4A</h2>
<pre class="r"><code>filepath &lt;- file.path(pathname, &quot;extdata/S4A_CSSII_ascr5.csv&quot;)
strains = c(&quot;N2&quot;,&quot;CSSII&quot;)

CSSII.C3 &lt;- read.csv(filepath, header =TRUE) %&gt;% dauergut::format_dauer(p.dauer = &quot;exclude&quot;) %&gt;% mutate(pheromone = factor(pheromone, levels = c(&quot;control&quot;,&quot;C3&quot;)))

lm &lt;- CSSII.C3 %&gt;% lm(., formula = pct ~ genotype*pheromone)
contrasts&lt;-dauergut::dunnett_contrasts(lm, ref.index = 1, factor  = &quot;genotype&quot;, interaction = &quot;pheromone&quot;)

plot.contrasts&lt;-c(rep(&quot;&quot;,4))

(p&lt;-ggplot(CSSII.C3, aes(x=pheromone)) +
  stat_summary(aes(y=pct, group=day), colour = &quot;black&quot;, fun.y = mean, geom = &quot;line&quot;, alpha = 0.2, linetype = 2) +
  add.median.dauer() +
  #add.Bayes.CI() +
  geom_dotplot(aes(y=pct, colour = pheromone, fill=pheromone),binwidth=.015, binaxis=&quot;y&quot;, position=&quot;dodge&quot;, stackdir=&quot;center&quot;, size =.3) +
    scale_color_manual( values = c(&quot;black&quot;,&quot;blue&quot;)) + 
    scale_fill_manual(values = c(&quot;black&quot;,&quot;blue&quot;)) +
  labs(title = &quot;Pheromone-induced dauer is not altered in CSSII&quot;,
           y = &quot;proportion dauer&quot;,
           x = &quot;pheromone&quot;) +
  facet_grid(.~genotype, switch=&quot;both&quot;) +
  #scale_colour_manual(values = c(&quot;black&quot;, &quot;#FF9933&quot;)) +
    #scale_fill_manual(values = c(&quot;black&quot;, &quot;#FF9933&quot;)) +
      scale_y_continuous(breaks=c(0,0.25,0.5,0.75, 1.0)) +
      scale_x_discrete(labels=function(x) sub(&quot; &quot;,&quot;\n&quot;,x,fixed=TRUE)) +
  #geom_text(data = plot.contrasts.H0, aes(x=2, label = prange, y = 1.075, group = NULL), size = 4) +
    stat_summary(aes(x=as.numeric(as.factor(pheromone)) + 0.3, y=-0.05),
                   fun.data = fun_length, geom = &quot;text&quot;, size = 3) +
  theme_my_classic + 
  theme(axis.title.x = element_blank(),
        axis.text.x = element_blank(),
        strip.text.x = element_text(size=10)))</code></pre>
<p><img src="Figure_7_files/figure-html/CSSII%20pheromone-1.png" width="672" /></p>
<p><strong>Figure S4A</strong><br />
Pheromone-induced dauer formation is not reduced in the CSSII strain. Each dot indicates the average number of dauers formed in a single assay. Horizontal bar indicates median. Numbers in parentheses below indicate the number of independent experiments with at least 40 animals each.</p>
</div>
<div id="c" class="section level2">
<h2>7C</h2>
<pre class="r"><code>filepath&lt;-file.path(pathname, &quot;extdata/7C_ewIRdata_allbinary.csv&quot;)

strains=c(&quot;N2&quot;,&quot;CB4856&quot;,&quot;ewIR19&quot;,&quot;ewIR20&quot;,
          &quot;ewIR21&quot;,&quot;ewIR22&quot;,&quot;ewIR23&quot;,&quot;ewIR24&quot;,
          &quot;ewIR25&quot;,&quot;ewIR26&quot;,&quot;ewIR27&quot;)

ewIR&lt;-read.csv(filepath, header=TRUE) %&gt;% format_dauer(p.dauer = &quot;exclude&quot;) %&gt;% dplyr::filter(n &gt; 9)


ewIR.nocont&lt;-subset(ewIR, !(genotype %in% c(&quot;N2&quot;,&quot;CB4856&quot;)))
mapping&lt;-read.csv(file.path(pathname, &#39;extdata/7C_chrII.markers.concise.csv&#39;))
mapping.nocont&lt;-subset(mapping, !(genotype %in% c(&quot;N2&quot;,&quot;CB4856&quot;)))
mapping.nocont$other.loci&lt;-NULL
ewIR&lt;-merge(ewIR, mapping, by=&quot;genotype&quot;, all=TRUE)
ewIR.nocont&lt;-merge(ewIR.nocont, mapping.nocont, by=&quot;genotype&quot;, all=TRUE)</code></pre>
<pre class="r"><code>#glmm.nest = glmer(data = ewIR, cbind(dauer,non) ~ genotype + (1|day/plateID), family=&quot;binomial&quot;,control=glmerControl(optimizer=&quot;bobyqa&quot;))
lm &lt;- ewIR %&gt;% dauer_ANOVA()
contrasts&lt;-dauergut::tukey_contrasts(lm, &quot;genotype&quot;)
stan.glmm &lt;- ewIR %&gt;% run_dauer_stan

mixed&lt;- stan.glmm %&gt;% getStan_CIs(type = &quot;dauer&quot;)
plot.contrasts&lt;-c(&quot;&quot;,contrasts$prange[1:10])
# get contrasts compared to ewIR25
#plot.contrasts.2&lt;-c(&quot;&quot;,&quot;&quot;,contrasts$prange[c(25,32,38,43,47,50)], &quot;&quot;, contrasts$prange[53:54]) 
plot.contrasts.2 &lt;- c(rep(&quot;&quot;,9),contrasts$prange[53:54])

labels &lt;- c(&quot;N2&quot;,&quot;CB4856&quot;,&quot;WN219&quot;,&quot;WN220&quot;,
          &quot;WN221&quot;,&quot;WN222&quot;,&quot;WN223&quot;,&quot;WN224&quot;,
          &quot;WN225&quot;,&quot;WN226&quot;,&quot;WN227&quot;)

(p&lt;-dauergut::plot_CIs(ewIR, title=&#39;CB4856 27º dauer formation QTL lies on ChrII&#39;, plot.contrasts, plot.contrasts.2 = plot.contrasts.2,ypos = 1.075, offset = 0.075, type = &quot;dauer&quot;, labels = labels))</code></pre>
<p><img src="Figure_7_files/figure-html/glmm%20ewIR%20strains-1.png" width="960" /></p>
<pre class="r"><code>img.path &lt;- file.path(pathname, &quot;figures/7B_ewIR_map.png&quot;)

include_graphics(img.path)</code></pre>
<p><img src="figures/7B_ewIR_map.png" width="425" /></p>
<p>7C. Chromosome near isogenic lines (NILs) were tested for 27º dauer formation. Each strain contains a segment of CB4856 DNA (blue, right) introgressed into an N2 background. EwIR25 contains a QTL which reduces dauer formation. ewIR27 contains an additional QTL that increases dauer formation. Diagram shows putative location of these QTL for decreased (blue) or increased (red) dauer formation. Box and scatter plot show raw data. Blue dot and bars show marginal means and 95% confidence interval, respectively, generalized linear mixed effects model using strain as a predictor (see methods). Black p-values reflect Tukey post-hoc comparision to N2, red p-values indicate comparison to ewIR25. n≥6 assays on ≥3 separate days.</p>
</div>
<div id="e" class="section level2">
<h2>7E</h2>
<pre class="r"><code>file&lt;-&#39;7E_MONIL_allbinary.csv&#39;
filepath&lt;-file.path(pathname, &quot;extdata&quot;, file)
strains&lt;-c(&quot;N2&quot;,&quot;CB4856&quot;,&quot;NIL10&quot;,&quot;NIL59&quot;,
                               &quot;NIL76&quot;,&quot;NIL78&quot;)

MONIL&lt;-read.csv(filepath, header=TRUE) %&gt;% format_dauer(p.dauer = &quot;exclude&quot;)
  # mutate(genotype = factor(genotype, levels = strains),
  #        pct = as.numeric(paste(dauer/(dauer+non))),
  #        strainDate = interaction(genotype, day),
  #        plateID = interaction(strainDate, plate))

file2&lt;-&#39;7E_MONIL_mapping.csv&#39;
filepath2&lt;-file.path(pathname, &quot;extdata&quot;, file2)
mapping&lt;-read.csv(filepath2)
controls &lt;- c(&quot;N2&quot;, &quot;CB&quot;)

MONIL &lt;- merge(MONIL, mapping, by=&quot;genotype&quot;, all=TRUE) %&gt;% mutate(binII12680000 = factor(binII12680000, levels = controls), 
                 binII12790000 = factor(binII12790000, levels = controls),
                 binII12900000 = factor(binII12900000, levels = controls),
                 binII13050000 = factor(binII13050000, levels = controls),
                 other = factor(other, levels = controls))</code></pre>
<pre class="r"><code>glmm.nest = glmer(data = MONIL, cbind(dauer,non) ~ genotype + (1|day/plateID), family=&quot;binomial&quot;, control=glmerControl(optimizer=&quot;bobyqa&quot;))

lm &lt;- MONIL %&gt;% dauer_ANOVA()
stanlmer &lt;- MONIL %&gt;% run_dauer_stan
#per-locus glmm - loci are fixed effects
loci&lt;-paste(names(mapping[-1]), collapse=&quot; + &quot;)
rand&lt;-&quot;(1|day/plateID)&quot;
f&lt;-paste(&quot;cbind(dauer,non) ~&quot;,loci,&quot; + &quot;,rand)
f.rev &lt;- paste(&quot;cbind(dauer,non) ~&quot;,rev(loci),&quot; + &quot;,rand)
locus.glmm&lt;-glmer(f, data=MONIL, family=binomial, control=glmerControl(optimizer=&quot;bobyqa&quot;))
locus.rev.glmm &lt;-glmer(f.rev, data=MONIL, family=binomial, control=glmerControl(optimizer=&quot;bobyqa&quot;))

# #bootstrapping
# # Not run:
# nsim=100 # takes ~12 seconds per iteration
# PBmod&lt;-drop1(locus.glmer,test=&quot;user&quot;,sumFun=PBSumFun)
# PBmod
# # end (**Not run**)</code></pre>
<pre class="r"><code>contrasts&lt;-dauergut::tukey_contrasts(glmm.nest, &quot;genotype&quot;)
mixed &lt;- stanlmer %&gt;% getStan_CIs(type = &quot;dauer&quot;)
plot.contrasts&lt;-c(&quot;&quot;,contrasts$prange[1:5])

(p&lt;-plot_CIs(MONIL, title=&#39;A ~100kb interval defines QTL1&#39;, plot.contrasts, ypos = 0.8, type = &quot;dauer&quot;, labels = strains))</code></pre>
<p><img src="Figure_7_files/figure-html/MONIL%20plot-1.png" width="672" /></p>
<pre class="r"><code>img.path &lt;- file.path(pathname, &quot;figures/7D_QTL1.png&quot;)

knitr::include_graphics(img.path)</code></pre>
<p><img src="figures/7D_QTL1.png" width="430" /></p>
<p><strong>Figure 7E</strong> (Top) Dauers formed by sub-NILs at 27°C. Each dot indicates the proportion of dauers formed in a single assay. Horizontal bar indicates median. Light gray thin and thick vertical bars at right indicate Bayesian 95% and 75% credible intervals, respectively. Numbers in parentheses below indicate the number of independent assays with at least 11 animals each. Shown P-values are with respect to the Bristol N2 strain unless indicated otherwise; <code>***</code> - different from Bristol N2 at P&lt;0.001 (ANOVA with Tukey-type multivariate-t post-hoc adjustment) (Bottom) Cartoon depicting CB4856 introgression breakpoint in NIL strains. Segments in blue indicate regions of CB4856 DNA in an otherwise Bristol N2 genetic background. Yellow shaded region indicates likely position of QTL1. Horizontal axis indicates physical position in megabases (Mb).</p>
</div>
<div id="f" class="section level2">
<h2>7F</h2>
<pre class="r"><code>strains&lt;- c(&quot;N2&quot;,&quot;CB4856&quot;,&quot;QTL1&quot;,&quot;QTL2&quot;)
days &lt;- c(&quot;3_10_17&quot;, &quot;3_17_17&quot;, &quot;4_6_17&quot;)
QTL.roam&lt;-read.csv(file.path(pathname, &quot;extdata/6B_7F_roam_wild_isolates.csv&quot;)) %&gt;%
  dplyr::filter(genotype %in% strains &amp; date %in% days) %&gt;%
  mutate(genotype = factor(genotype, levels = strains),
         strainDate = interaction(genotype, date, drop=TRUE),
         plateID = interaction(strainDate,plate, drop=TRUE),
         total.boxes = 186)


lm &lt;- lm(n_entries ~ genotype, data = QTL.roam)

QTL.roam %&lt;&gt;% dauergut::flag_outliers(df = ., lin.mod = lm, threshold = 4, noplot = TRUE)

glmm.nest = glmer(data=QTL.roam, n_entries ~ genotype + (1|date/plateID), family=&quot;poisson&quot;)

stan.glmm &lt;- stan_glmer(data = QTL.roam[QTL.roam$outlier.status == FALSE,], 
                        formula = n_entries ~ genotype + (1|date/plateID), family = &quot;poisson&quot;,
                        iter = 4000, adapt_delta = 0.99)

lm &lt;- update(lm, data = QTL.roam[QTL.roam$outlier.status == FALSE,])

contrasts&lt;-dauergut::tukey_contrasts(lm, &quot;genotype&quot;)
mixed&lt;-stan.glmm %&gt;% getStan_CIs(type = &quot;roam&quot;)
plot.contrasts&lt;-c(&quot;&quot;,contrasts$prange[1:3])
plot.contrasts.2 &lt;- c(&quot;&quot;,&quot;&quot;,contrasts$prange[4:5])

(p&lt;-dauergut::plot_CIs(QTL.roam, title=&#39;Foraging behavior is not correlated with Hid phenotypes&#39;, plot.contrasts=plot.contrasts, plot.contrasts.2 = plot.contrasts.2, ypos = 800, offset = 50, type = &quot;grid&quot;, labels = strains))</code></pre>
<p><img src="Figure_7_files/figure-html/roam%20QTL-1.png" width="672" /></p>
<p><strong>Figure 7F</strong><br />
Exploratory behavior of indicated strains. Each purple dot is data from a single animal. Median is indicated by a horizontal line; error bars are quartiles. Light gray thin and thick vertical bars at right indicate Bayesian 95% and 75% credible intervals, respectively. Numbers in parentheses below indicate the total number of animals examined in 3 independent assay days. P-values shown are with respect to N2; <span style="color:red"><code>***</code></span> - different from CB4856 at P&lt;0.001 (ANOVA with Tukey-type multivariate-t post-hoc adjustments).</p>
</div>
<div id="s4c" class="section level2">
<h2>S4C</h2>
<pre class="r"><code>img.path &lt;- file.path(pathname, &quot;figures/S4B_rict-1_locus.png&quot;)
include_graphics(img.path)</code></pre>
<p><img src="figures/S4B_rict-1_locus.png" width="1032" /></p>
<pre class="r"><code>strains&lt;-c(&quot;N2&quot;, &quot;ewIR25&quot;, &quot;NK&quot;, &quot;NK_DG&quot;)
foods &lt;- &quot;OP50&quot;

NKDG&lt;-read.csv(file.path(pathname, &quot;extdata&quot;, &quot;S4C_NKDG_allbinary.csv&quot;), header=TRUE) %&gt;% format_dauer(p.dauer = &quot;non&quot;) #including partial/pd as non-dauer due to excess zeros. 


lm &lt;- NKDG %&gt;% dauer_ANOVA()
#stan
stan.glmm &lt;- NKDG %&gt;% run_dauer_stan

contrasts&lt;-dauergut::dunnett_contrasts(lm, ref.index = 1, &quot;genotype&quot;)
mixed&lt;-stan.glmm %&gt;% getStan_CIs(type = &quot;dauer&quot;)
plot.contrasts&lt;-c(&quot;&quot;,contrasts$prange[1:3])

labels &lt;- c(&quot;N2&quot;, &quot;WN225&quot;, &quot;rict-1 [N1174K]&quot;, &quot;rict-1 [N1174K, D1462G]&quot;) %&gt;% stringr::str_wrap(width = 10)

(p&lt;-dauergut::plot_CIs(NKDG, title=&#39;mTORC2 prevents high temperature dauer formation&#39;, plot.contrasts, ypos = 1.075, type = &quot;dauer&quot;, labels = labels))</code></pre>
<p><img src="Figure_7_files/figure-html/NKDG%20CRISPR-1.png" width="672" /></p>
<p><strong>Figure S4C</strong><br />
Dauers formed at 27°C by N2, the WN225 NIL, and strains in which the indicated polymorphisms in CB4856 rict-1 coding sequences have been introduced into N2 rict-1 sequences via gene editing. Each dot indicates the average number of dauers formed in a single assay. Horizontal bar indicates median. Light-grey thick and thin vertical bars indicate Bayesian 95% and 75% credible intervals, respectively. Numbers in parentheses below indicate the number of independent experiments with at least 40 animals each. P-values are relative to N2 (ANOVA with Dunnett-type multivariate-t post-hoc adjustment).</p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
